qiita_db/software.py

```diff
@@ -413,31 +484,37 @@ def exists(cls, command, **kwargs):
         -------
         bool
             Whether if the parameter set exists in the given command
+
+        Raises
+        ------
+        qiita_db.exceptions.QiitaDBError
+            - If there are missing parameters for the given command
+            - If there are extra parameters in `kwargs` than for the given
+              command
         """
         with qdb.sql_connection.TRN:
-            table = command.parameters_table
+            command_params = set(command.optional_parameters)
+            user_params = set(kwargs)
 
-            db_cols = set(qdb.util.get_table_cols(table))
-            db_cols.remove("param_set_name")
-            db_cols.remove("parameters_id")
-            missing = db_cols.difference(kwargs)
+            missing_in_user = command_params - user_params
+            extra_in_user = user_params - command_params
 
-            if missing:
+            if missing_in_user or extra_in_user:
                 raise qdb.exceptions.QiitaDBError(
-                    "Missing parameters for command %s: %s"
-                    % (command.name, ', '.join(missing)))
+                    "The given set of parameters do not match the ones for "
+                    "the command.\nMissing parameters: %s\n"
+                    "Extra parameters: %s\n"
+                    % (', '.join(missing_in_user), ', '.join(extra_in_user)))
 
-            extra = set(kwargs).difference(db_cols)
-            if extra:
-                raise qdb.exceptions.QiitaDBError(
-                    "Extra parameters for command %s: %s"
-                    % (command.name, ', '.join(extra)))
+            sql = """SELECT parameter_set
+                     FROM qiita.default_parameter_set
+                     WHERE command_id = %s"""
+            qdb.sql_connection.TRN.add(sql, [command.id])
+            for p_set in qdb.sql_connection.TRN.execute_fetchflatten():
+                if p_set == kwargs:
```

Comment:
Yes, they should both be a dict of `{parameter_name: value}`, and this is checking if the given combination of values already exists in the system.